rdtools/analysis_chains.py

```diff
@@ -0,0 +1,829 @@
+
+        Parameters
+        ----------
+        post_filter : pandas.Series
+            Time series filtered by boolean output from self.filter
+        '''
+        post_filter_length = post_filter.index[-1] - post_filter.index[0]
+        if post_filter.empty or post_filter_length < pd.Timedelta('730d'):
+            raise ValueError(
+                "Less than two years of data left after filtering")
+
+    def aggregate(self, energy_normalized, insolation):
+        '''
+        Return insolation-weighted normalized PV energy and the associated aggregated insolation
+
+        Parameters
+        ----------
+        normalized : pandas.Series
+            Time series of normalized PV energy
+        insolation : pandas.Series
+            Time Series of insolation associated with each `normalized` point
+
+        Returns
+        -------
+        pandas.Series
+            Insolation-weighted aggregated normalized PV energy
+        pandas.Series
+            Aggregated insolation
+        '''
+        aggregated = aggregation.aggregation_insol(
+            energy_normalized, insolation, self.aggregation_freq)
+        aggregated_insolation = insolation.resample(
+            self.aggregation_freq).sum()
+
+        return aggregated, aggregated_insolation
+
+    def yoy_degradation(self, energy_normalized, **kwargs):
+        '''
+        Perform year-on-year degradation analysis on insolation-weighted
+        aggregated energy yield.
+
+        Parameters
+        ----------
+        energy_normalized : pandas.Series
+            Time Series of insolation-weighted aggregated normalized PV energy
+        kwargs :
+            Extra parameters passed to degradation.degradation_year_on_year()
+
+        Returns
+        -------
+        dict
+            Year-on-year degradation results with keys:
+            'p50_rd' : The median year-on-year degradation rate
+            'rd_confidence_interval' : lower and upper bounds of degradation
+                                       rate confidence interval as a list
+            'calc_info': Dict of detailed results
+                         (see degradation.degradation_year_on_year() docs)
+        '''
+        self._filter_check(energy_normalized)
+        yoy_rd, yoy_ci, yoy_info = degradation.degradation_year_on_year(
+            energy_normalized, **kwargs)
+
+        yoy_results = {
+            'p50_rd': yoy_rd,
+            'rd_confidence_interval': yoy_ci,
+            'calc_info': yoy_info
+        }
+
+        return yoy_results
+
+    def srr_soiling(self, energy_normalized_daily, insolation_daily, **kwargs):
+        '''
+        Perform stochastic rate and recovery soiling analysis.
+
+        Parameters
+        ---------
+        energy_normalized_daily : pandas.Series
+            Time Series of insolation-weighted aggregated normalized PV energy
+        insolation_daily : pandas.Series
+            Time Series of insolation, aggregated at same level as `aggregated`
```

Comment:
`same level as aggregated` -- change aggregated to whatever the updated parameter name lands on